zap/src/main/java/org/zaproxy/zap/model/Context.java

```diff
@@ -604,59 +597,30 @@ public void run() {
 
     private void restructureSiteTreeEventHandler() {
         log.debug("Restructure site tree for context: " + this.getName());
+        // long time = System.nanoTime();
         List<SiteNode> nodes = this.getTopNodesInContextFromSiteTree();
         for (SiteNode sn : nodes) {
-            checkNode(sn);
+            updateNode(sn);
         }
+        // log.debug("restructuring took {} nanoseconds", System.nanoTime() - time);
```

Comment:
While the updated nodes use the correct name from the variant, the sites tree is updated a little more slowly than before. This is because it updates ALL the nodes now. Any suggestions to improve the performance are welcome here...

Reply:
@ricekot when does the slowdown occur? Whenever the Sites tree is updated or ??

Reply:
Whenever a context is modified and `Context#restructureSiteTree` is called.

Reply:
how significant is the slowdown?